src/local_scheduler/local_scheduler.cc

```diff
@@ -59,11 +59,79 @@ void print_resource_info(const LocalSchedulerState *state,
 #endif
 }
 
+bool check_dynamic_resources(LocalSchedulerState *state,
+                             double num_cpus,
+                             double num_gpus) {
+  if (num_cpus > 0 && state->dynamic_resources[ResourceIndex_CPU] < num_cpus) {
```

Comment:
I omitted it because `num_cpus` is allowed to drop below `0` but `num_gpus` is not.